src/components/pages/deployments/flyouts/deploymentNew/deploymentNew.js

```diff
@@ -0,0 +1,300 @@
+    const selectedDeviceGroupId = e.target.value.value;
+    const selectedDeviceGroup = deviceGroups.find(deviceGroup => deviceGroup.id === selectedDeviceGroupId)
+    fetchDevices(selectedDeviceGroup.conditions);
+  }
+
+  formControlChange = () => {
+    if (this.state.changesApplied) {
+      this.setState({ changesApplied: false });
+    }
+  }
+
+  toPackageSelectOption = ({ id, name }) => ({ label: name, value: id });
+
+  toDeviceGroupSelectOption = ({ id, displayName }) => ({ label: displayName, value: id });
+
+  render() {
+    const {
+      t,
+      onClose,
+      createIsPending,
+      createError,
+      packagesPending,
+      packagesError,
+      packages,
+      deviceGroups,
+      devicesPending,
+      devicesError
+    } = this.props;
+    const {
+      name,
+      packageType,
+      deviceGroupId,
+      packageId,
+      priority,
+      targetedDeviceCount,
+      changesApplied,
+      edgePackageSelected,
+    } = this.state;
+
+    // Validators
+    const requiredValidator = (new Validator()).check(Validator.notEmpty, t('deployments.flyouts.new.validation.required'));
+
+    // Links
+    this.packageTypeLink = this.linkTo('packageType').map(({ value }) => value).withValidator(requiredValidator);
+    this.nameLink = this.linkTo('name').withValidator(requiredValidator);
+    this.deviceGroupIdLink = this.linkTo('deviceGroupId').map(({ value }) => value).withValidator(requiredValidator);
+    this.priorityLink = this.linkTo('priority')
+      .check(Validator.notEmpty, () => this.props.t('deployments.flyouts.new.validation.required'))
+      .check(val => !isNaN(val), t('deployments.flyouts.new.validation.nan'));
+    this.packageIdLink = this.linkTo('packageId').map(({ value }) => value).withValidator(requiredValidator);
+
+    const isPackageTypeSelected = packageType !== '';
+    const isDeviceGroupSelected = deviceGroupId !== '';
+    const packageOptions = packages.map(this.toPackageSelectOption);
+    const deviceGroupOptions = deviceGroups.map(this.toDeviceGroupSelectOption);
+    const typeOptions = packageTypeOptions.map(value => ({
+      label: t(`deployments.typeOptions.${value.toLowerCase()}`),
+      value
+    }));
+    const completedSuccessfully = changesApplied && !createError && !createIsPending;
+    const deviceFetchSuccessful = isDeviceGroupSelected && !devicesError && !devicesPending;
+
+    return (
+      <Flyout>
+        <FlyoutHeader>
+          <FlyoutTitle>{t('deployments.flyouts.new.title')}</FlyoutTitle>
+          <FlyoutCloseBtn onClick={onClose} />
+        </FlyoutHeader>
+        <FlyoutContent className="new-deployment-content">
+          <form className="new-deployment-form" onSubmit={this.apply}>
+            <FormGroup className="new-deployment-formGroup">
+              <FormLabel isRequired="true">{t('deployments.flyouts.new.name')}</FormLabel>
+              {
+                !completedSuccessfully && <FormControl
```

Comment:
The spacing here feels odd. Put the opening tag starting on the next line. (same thing several times below)

Reply:
done